examples/platforms/samr21/radio.c

```diff
@@ -0,0 +1,611 @@
+    {
+        setChannel(aChannel);
+
+        radioRxEnable();
+
+        sState = OT_RADIO_STATE_RECEIVE;
+
+        error = OT_ERROR_NONE;
+    }
+
+    return error;
+}
+
+otError otPlatRadioTransmit(otInstance *aInstance, otRadioFrame *aFrame)
+{
+    (void)aInstance;
+
+    otLogDebgPlat(sInstance, "Radio transmit");
+
+    otError error = OT_ERROR_INVALID_STATE;
+
+    if (sState == OT_RADIO_STATE_RECEIVE)
+    {
+        uint8_t frame[OT_RADIO_FRAME_MAX_SIZE + 1];
+
+        setChannel(aFrame->mChannel);
+        setTxPower(aFrame->mPower);
+
+        frame[0] = aFrame->mLength - IEEE802154_FCS_SIZE;
+        memcpy(frame + 1, aFrame->mPsdu, aFrame->mLength);
+
+        PHY_DataReq(frame);
+
+        otPlatRadioTxStarted(aInstance, aFrame);
+
+        sState = OT_RADIO_STATE_TRANSMIT;
+
+        error = OT_ERROR_NONE;
+    }
+
+    return error;
+}
+
+otRadioFrame *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    return &sTransmitFrame;
+}
+
+int8_t otPlatRadioGetRssi(otInstance *aInstance)
+{
+    return sMaxRssi;
+}
+
+otRadioCaps otPlatRadioGetCaps(otInstance *aInstance)
+{
+    return OT_RADIO_CAPS_ENERGY_SCAN | OT_RADIO_CAPS_TRANSMIT_RETRIES |
+           OT_RADIO_CAPS_ACK_TIMEOUT;
+}
+
+bool otPlatRadioGetPromiscuous(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    return sPromiscuous;
+}
+
+void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
+{
+    (void)aInstance;
+
+    sPromiscuous = aEnable;
+}
+
+void otPlatRadioEnableSrcMatch(otInstance *aInstance, bool aEnable)
+{
+    (void)aInstance;
+    (void)aEnable;
+}
+
+otError otPlatRadioAddSrcMatchShortEntry(otInstance *aInstance, const uint16_t aShortAddress)
+{
+    return OT_ERROR_NOT_IMPLEMENTED;
+}
+
+otError otPlatRadioAddSrcMatchExtEntry(otInstance *aInstance, const otExtAddress *aExtAddress)
+{
+    return OT_ERROR_NOT_IMPLEMENTED;
+}
+
+otError otPlatRadioClearSrcMatchShortEntry(otInstance *aInstance, const uint16_t aShortAddress)
+{
+    return OT_ERROR_NOT_IMPLEMENTED;
+}
+
+otError otPlatRadioClearSrcMatchExtEntry(otInstance *aInstance, const otExtAddress *aExtAddress)
+{
+    return OT_ERROR_NOT_IMPLEMENTED;
+}
+
+void otPlatRadioClearSrcMatchShortEntries(otInstance *aInstance)
+{
+    (void)aInstance;
+}
+
+void otPlatRadioClearSrcMatchExtEntries(otInstance *aInstance)
+{
+    (void)aInstance;
+}
+
+otError otPlatRadioEnergyScan(otInstance *aInstance, uint8_t aScanChannel, uint16_t aScanDuration)
+{
+    (void)aInstance;
+
+    sScanStartTime = otPlatAlarmMilliGetNow();
+    sScanDuration = aScanDuration;
+
+    sMaxRssi = PHY_EdReq();
+
+    sStartScan = true;
+
+    return OT_ERROR_NONE;
+}
+
+void otPlatRadioSetDefaultTxPower(otInstance *aInstance, int8_t aPower)
+{
+    (void)aInstance;
+
+    otLogDebgPlat(sInstance, "Radio set default TX power: %d", aPower);
+
+    setTxPower(aPower);
+}
+
+int8_t otPlatRadioGetReceiveSensitivity(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    return -100;
```

Comment:
Can this be defined as a `#define` constant?

Reply:
Sure, I will add define. And probably it should be set to -99 according to spec:
Integrated Ultra Low Power Transceiver for 2.4GHz ISM Band:
    -99dBm RX Sensitivity; TX Output Power up to +4dBm